### PR TITLE
15876 fix time random casting with without timezone

### LINKED ISSUE #15876: `TIME` are randomly cast with/without timezone depending on millisecond values
When doing a native query to cast a datetime to time, the results are inconsistent. Some of the rows have incorrect time, in my particular case it's -7 hours. See here: https://i.imgur.com/RbKGdTq.png. For sanity, here's the same query ran in PgAdmin: https://i.imgur.com/3KA4bHT.png

Using Metabase cloud, not self hosted.

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by losrebellos**
Fixes #15876.

- [x] Fix for the `parseTime` method
- [x] New tests added for the `parseTime` method
- [x] Tests created for the `formatTime` method

__Before__

Using the repro steps from https://github.com/metabase/metabase/issues/15876#issuecomment-830210005.
image

__After__

All the dates are normalised regardless of the milliseconds.
image

### CODE REVIEW

File: frontend/src/metabase/lib/time.js [hunk 1]
```diff
@@ -103,16 +103,15 @@ export function parseTime(value) {
   if (moment.isMoment(value)) {
     return value;
   } else if (typeof value === "string") {
-    return moment.parseZone(value, [
-      "HH:mm:SS.sssZZ",
-      "HH:mm:SS.sss",
+    return moment(value, [
```
Comment (losrebellos) [inline frontend/src/metabase/lib/time.js:106]: Removing timezone from `moment` time parsing.

File: frontend/test/metabase/lib/time.unit.spec.js [hunk 2]
```diff
@@ -59,19 +59,26 @@ describe("time", () => {
...
-    it("parse timezones", () => {
-      const result = parseTime("01:02:03.456+07:00");
-
-      expect(moment.isMoment(result)).toBe(true);
-      expect(result.format("h:mm A")).toBe("1:02 AM");
-    });
-
-    it("parse time without seconds", () => {
-      const result = parseTime("01:02");
+    const PARSE_TIME_TESTS = [
```
Comment (losrebellos) [inline frontend/test/metabase/lib/time.unit.spec.js:62]: New tests added for `parseTime` method.

File: frontend/test/metabase/lib/time.unit.spec.js [hunk 3]
```diff
@@ -59,19 +59,26 @@ describe("time", () => {
   });
 
   describe("parseTime", () => {
```
Comment (losrebellos) [inline frontend/test/metabase/lib/time.unit.spec.js:61]: New tests added for `parseTime` method.

File: frontend/test/metabase/lib/formatting.unit.spec.js [hunk 4]
```diff
@@ -522,6 +523,28 @@ describe("formatting", () => {
     });
   });
 
+  describe("formatTime", () => {
```
Comment (losrebellos) [inline frontend/test/metabase/lib/formatting.unit.spec.js:526]: Tests created for `formatTime` method.

File: frontend/src/metabase/lib/time.js [hunk 5]
```diff
@@ -103,16 +103,15 @@ export function parseTime(value) {
...
+      "HH:mm:ss.sss[Z]",
```
Comment (losrebellos) [inline frontend/src/metabase/lib/time.js:107]: `HH:mm:ss.sss[Z]` adds support to more milliseconds format parsing.

File: frontend/src/metabase/lib/time.js [hunk 6]
```diff
@@ -103,16 +103,15 @@ export function parseTime(value) {
   if (moment.isMoment(value)) {
     return value;
   } else if (typeof value === "string") {
-    return moment.parseZone(value, [
-      "HH:mm:SS.sssZZ",
-      "HH:mm:SS.sss",
```
Comment (losrebellos) [inline frontend/src/metabase/lib/time.js:108]: Remove duplicates.

**Review by flamber — COMMENTED** [review]
This seems to be working as expected - awesome :tada:
Can you also enable the Cypress test #21316 (I'm unsure if it needs adjustment depending on instance timezone).

**Comment by losrebellos** [comment]
> This seems to be working as expected - awesome 🎉 Can you also enable the Cypress test #21316 (I'm unsure if it needs adjustment depending on instance timezone).

Test is enabled now and all green.

**Review by flamber — APPROVED** [review]
I don't know Moment very well, but this seems correct.

**Review by daltojohnso — APPROVED** [review]
Interesting, wasn't aware of the brackets in the moment formatting string. Tested and it works

**Review by ariya — APPROVED** [review]
Great stuff, let's ship this 🚀

**Pull request merged**